src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -453,6 +451,49 @@ impl<'index, D: Decoder> IndexReader<'index, D> {
 
         Ok(Some(result))
     }
+
+    /// Skip forward to the block containing the given document ID. Returns false if the end of the
+    /// index was reached and true otherwise.
+    pub fn skip_to(&mut self, doc_id: t_docId) -> bool {
+        if self.current_block.last_doc_id >= doc_id {
+            // We are already in the correct block
+            return true;
+        }
+
+        // SAFETY: it is safe to unwrap because we checked that the blocks are not empty when
```

Comment:
The garbage collector will always ensure there is at least one block left?

Reply:
Yes, the current GC does. It will also be impossible to set `current_block` if all the blocks are cleaned out